src/precice/impl/WatchIntegral.cpp

```diff
@@ -105,10 +105,10 @@ void WatchIntegral::exportIntegralData(
   }
 }
 
-Eigen::VectorXd WatchIntegral::calculateIntegral(const mesh::PtrData &data) const
+Eigen::VectorXd WatchIntegral::calculateIntegral(double time, const mesh::PtrData &data) const
 {
   int                    dim    = data->getDimensions();
-  const Eigen::VectorXd &values = data->timeStepsStorage().last().sample.values;
+  const Eigen::VectorXd &values = data->timeStepsStorage().sample(time);
```

Comment:
This point in time should always exist as a sample, correct?

It would be nice for such cases to have a function that makes sure that there is no interpolation taking place.

Reply:
The implementation of `sample(...)` already checks if a value exists in the storage. If no value exists in the storage interpolation basically the fall-back IIRC. Introducing an additional argument like `sample(..., errorOnInterpolation=false)` would allow us to enforce a stricter behavior in tests. Good idea, but let's not open this box here please :pray: